plugins/Sf2Player/Sf2Player.cpp

```diff
@@ -622,18 +663,19 @@ void Sf2Instrument::playNote( NotePlayHandle * _n, sampleFrame * )
 	}
 
 	const f_cnt_t tfp = _n->totalFramesPlayed();
+	
+	int masterPitch = instrumentTrack()->useMasterPitchModel()->value() ? Engine::getSong()->masterPitch() : 0;
+	int baseNote = instrumentTrack()->baseNoteModel()->value();
+	int midiNote = _n->midiKey() - baseNote + DefaultBaseKey + masterPitch;
 
-	if( tfp == 0 )
+	// out of range?
+	if( midiNote <= 0 || midiNote >= 128 )
```

Comment:
Not originally in scope, but while you're touching this bit of code, you could fix the range check to allow playing C-1:
```suggestion
	if (midiNote < 0 || midiNote >= 128)
```